code/modules/mob/living/carbon/xenomorph/castes/drone/abilities_drone.dm

```diff
@@ -58,3 +58,67 @@
 // ***************************************
 /datum/action/xeno_action/create_jelly/slow
 	cooldown_timer = 45 SECONDS
+
+// ***************************************
+// *********** Sow
+// ***************************************
+/datum/action/xeno_action/sow
+	name = "Sow"
+	action_icon_state = "place_trap"
+	mechanics_text = "Sow the seeds of an alien plant."
+	plasma_cost = 500
+	cooldown_timer = 1 MINUTES
+	use_state_flags = XACT_USE_LYING
+	keybind_signal = COMSIG_XENOABILITY_DROP_PLANT
+	alternate_keybind_signal = COMSIG_XENOABILITY_CHOOSE_PLANT
+
+/datum/action/xeno_action/sow/can_use_action(silent = FALSE, override_flags)
+	. = ..()
+	var/turf/T = get_turf(owner)
+
+	if(!(locate(/obj/effect/alien/weeds) in T))
+		if(!silent)
+			to_chat(owner, span_warning("Only weeds are fertile enough for our plants!"))
+		return FALSE
+
+	if((locate(/obj/structure/xeno/plant) in T))
```

Comment:
If you meant the weed check I tested and it isn't in check_alien_construction. 
Otherwise if it's the plant one I added so the ability itself could recognize because you already had stuff growing here but I can remove it.
Unresolving